openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -299,171 +302,181 @@ def _uidb36_to_uidb64(uidb36):
+            # We have to make a copy of request.POST because it is a QueryDict object which is immutable until copied.
+            # We have to use request.POST because the password_reset_confirm method takes in the request and a user's
+            # password is set to the request.POST['new_password1'] field. We have to also normalize the new_password2
+            # field so it passes the equivalence check that new_password1 == new_password2
+            # In order to switch out of having to do this copy, we would want to move the normalize_password code into
+            # a custom User model's set_password method to ensure it is always happening upon calling set_password.
+            request.POST = request.POST.copy()
+            request.POST['new_password1'] = normalize_password(request.POST['new_password1'])
+            request.POST['new_password2'] = normalize_password(request.POST['new_password2'])
+
+            password = request.POST['new_password1']
 
-        # get the updated user
-        updated_user = User.objects.get(id=uid_int)
-        if 'is_account_recovery' in request.GET:
             try:
-                updated_user.email = updated_user.account_recovery.secondary_email
-                updated_user.account_recovery.delete()
-                # emit an event that the user changed their secondary email to the primary email
-                tracker.emit(
-                    SETTING_CHANGE_INITIATED,
-                    {
-                        "setting": "email",
-                        "old": user.email,
-                        "new": updated_user.email,
-                        "user_id": updated_user.id,
-                    }
+                validate_password(password, user=user)
+            except ValidationError as err:
+                # We have a password reset attempt which violates some security
+                # policy, or any other validation. Use the existing Django template to communicate that
+                # back to the user.
+                context = {
+                    'validlink': True,
+                    'form': None,
+                    'title': _('Password reset unsuccessful'),
+                    'err_msg': ' '.join(err.messages),
+                }
+                context.update(platform_name)
+                return TemplateResponse(
+                    request, 'registration/password_reset_confirm.html', context
                 )
-            except ObjectDoesNotExist:
-                log.error(
-                    u'Account recovery process initiated without AccountRecovery instance for user {username}'.format(
-                        username=updated_user.username
+
+            if 'is_account_recovery' in request.GET:
+                self.post_reset_login = True
+                response = self._process_password_reset_success(request, token, uidb64)
+            else:
+                response = self._process_password_reset_success(request, token, uidb64)
```

Comment:
Is there a reason why we're trying to process the form ourselves instead of just deferring to `PasswordResetConfirmView` like the previous functionality did?

Reply:
`form_valid` in `PasswordResetConfirmView` tries to save the form as soon as it receives and it fails in one of its base classes, we've to call `is_valid` before it gets saved.